base/src/main/java/com/smartdevicelink/managers/screen/choiceset/PreloadPresentChoicesOperation.java

```diff
@@ -0,0 +1,787 @@
+            cancelInteraction.setOnRPCResponseListener(new OnRPCResponseListener() {
+                @Override
+                public void onResponse(int correlationId, RPCResponse response) {
+                    if (response.getSuccess()) {
+                        DebugTool.logInfo(TAG, "Canceled the presented choice set " + ((response.getResultCode() == Result.SUCCESS) ? "successfully" : "unsuccessfully"));
+                    } else {
+                        DebugTool.logError(TAG, "Error canceling the presented choice set: " + correlationId + " with error: " + response.getInfo());
+                    }
+                }
+            });
+
+            if (internalInterface.get() != null) {
+                internalInterface.get().sendRPC(cancelInteraction);
+            } else {
+                DebugTool.logError(TAG, "Internal interface null - could not send cancel interaction for choice set");
+            }
+        } else {
+            DebugTool.logInfo(TAG, "Canceling a choice set that has not yet been sent to Core");
+            this.cancelTask();
+        }
+    }
+
+    //Present Helpers
+    void setSelectedCellWithId(Integer cellId) {
+        if (choiceSet.getChoices() != null && cellId != null) {
+            List<ChoiceCell> cells = choiceSet.getChoices();
+            for (int i = 0; i < cells.size(); i++) {
+                if (cells.get(i).getChoiceId() == cellId) {
+                    selectedCell = cells.get(i);
+                    selectedCellRow = i;
+                    return;
+                }
+            }
+        }
+    }
+
+    PerformInteraction getPerformInteraction() {
+        if (this.choiceSet == null) {
+            return new PerformInteraction();
+        }
+        PerformInteraction pi = new PerformInteraction(choiceSet.getTitle(), presentationMode, getChoiceIds());
+        pi.setInitialPrompt(choiceSet.getInitialPrompt());
+        pi.setHelpPrompt(choiceSet.getHelpPrompt());
+        pi.setTimeoutPrompt(choiceSet.getTimeoutPrompt());
+        pi.setVrHelp(choiceSet.getVrHelpList());
+        pi.setTimeout(choiceSet.getTimeout() * 1000);
+        pi.setInteractionLayout(getLayoutMode());
+        pi.setCancelID(cancelID);
+        return pi;
+    }
+
+    //Choice Uniqueness
+    void updateCellsBasedOnLoadedChoices() {
+        if (internalInterface.get().getProtocolVersion().getMajor() >= 7 && internalInterface.get().getProtocolVersion().getMinor() >= 1) {
+            addUniqueNamesToCells(cellsToUpload);
+        } else {
+            ArrayList<ChoiceCell> strippedCellsCopy = (ArrayList<ChoiceCell>) removeUnusedProperties(cellsToUpload);
+            addUniqueNamesBasedOnStrippedCells(strippedCellsCopy, cellsToUpload);
+        }
+        cellsToUpload.removeAll(loadedCells);
+    }
+
+    List<ChoiceCell> removeUnusedProperties(List<ChoiceCell> choiceCells) {
+        List<ChoiceCell> strippedCellsClone = cloneChoiceCellList(choiceCells);
+        //Clone Cells
```

Comment:
This is my fault from a previous PR as this is just copied from BaseChoiceSetManager.
```suggestion
        // Clone Cells
```